util/design/lib/OtpMemMap.py

```diff
@@ -165,19 +165,34 @@ def _validate_mmap(config):
 
     offset = 0
     num_part = 0
+    part_index = OrderedDict()
     for part in config["partitions"]:
-        num_part += 1
         _validate_part(part, offset, key_names)
 
+        if part['name'] in part_index:
+            log.error('Partition name {} is not unique'.format(part['name']))
+            exit(1)
+
         # Loop over items within a partition
+        num_items = 0
+        item_index = OrderedDict()
         for item in part["items"]:
             _validate_item(item, offset)
+            if item['name'] in item_index:
+                log.error('Item name {} is not unique'.format(item['name']))
+                exit(1)
             log.info("> Item {} at offset {} with size {}".format(
                 item["name"], offset, item["size"]))
             offset += check_int(item["size"])
+            item_index.setdefault(item['name'], num_items)
+            num_items += 1
 
         # Place digest at the end of a partition.
         if part["sw_digest"] or part["hw_digest"]:
+            if part["name"] + DIGEST_SUFFIX in item_index:
+                log.error('Digest name is not unique'.format(part["name"] + DIGEST_SUFFIX))
+                exit(1)
+            item_index.setdefault(part["name"] + DIGEST_SUFFIX, num_items)
```

Comment:
I'd probably suggest defining a variable for `part['name'] + DIGEST_SUFFIX` to avoid repeating it. Also, I probably wouldn't use `setdefault` (as above).

Reply:
yeah that's duplicated many times, let me replace that with a var